apm-agent-plugins/apm-vertx/apm-vertx-common/src/main/java/co/elastic/apm/agent/vertx/AbstractVertxWebHelper.java

```diff
@@ -78,7 +78,7 @@ protected void setRouteBasedTransactionName(Transaction<?> transaction, RoutingC
         if (!webConfiguration.isUsePathAsName()) {
             String path = routingContext.currentRoute().getPath();
             if (path != null) {
-                StringBuilder transactionName = transaction.getAndOverrideName(AbstractSpan.PRIO_HIGH_LEVEL_FRAMEWORK);
+                StringBuilder transactionName = transaction.getAndOverrideName(AbstractSpan.PRIO_HIGH_LEVEL_FRAMEWORK - 1);
```

Comment:
[question] why did this priority needed to be changed ? Does it means without this the quarkus transaction name is not the one we expect ?

Reply:
This is required to make the instrumentation work correctly with RestEasy: RestEasy uses JAX-RS annotations, which we already instrument. In the case of quarkus it however collides with the Vert.x router instrumentation, because both have the same priority, but the router comes first.
I think it makes sense to lower the vert.x router prio in general, because it can be expected that the router is actually used a tool to implement higher-level frameworks (such as JAX-RS).